dataloadgen.go

```diff
@@ -98,24 +99,27 @@ func (l *Loader[KeyT, ValueT]) Load(ctx context.Context, key KeyT) (ValueT, erro
 // This method should be used if you want one goroutine to make requests to many
 // different data loaders without blocking until the thunk is called.
 func (l *Loader[KeyT, ValueT]) LoadThunk(ctx context.Context, key KeyT) func() (ValueT, error) {
-	ctx, span := l.tracer.Start(ctx, "dataloadgen.load")
-	defer span.End()
 	l.mu.Lock()
 	defer l.mu.Unlock()
 	if it, ok := l.thunkCache[key]; ok {
 		return it
 	}
 
-	l.startBatch(ctx)
+	_, loadSpan := l.tracer.Start(ctx, "dataloadgen.load")
+	defer loadSpan.End()
+
+	l.startBatch()
 
 	l.batch.contexts = append(l.batch.contexts, ctx)
+
+	_, waitSpan := l.tracer.Start(ctx, "dataloadgen.wait")
```

Comment:
This still assumes there's always a tracer, so let's change that and get this shipped